### ISSUE TITLE
add data source MGI

### ISSUE THREAD

**Issue by nlwashington** [monarch-initiative/dipper#13]
add mgi data from remote db:

adhoc.informatics.jax.org:5432/mgd

we currently use the following views:
mgi_organism_acc_view mgi_organism_view gxd_genotype_view gxd_allelepair_view mrk_marker_view mgi_reference_allele_view bib_acc_view voc_annot_view gxd_genotype_summary_view voc_evidence_view all_allele_cellline_view voc_term_view all_allele_view all_allele_mutation_view prb_strain_view all_summary_view mrk_summary_view mgi_note_vocevidence_view acc_logicaldb_view mgi_note_strain_view prb_strain_acc_view prb_strain_summary_view prb_strain_marker_view

**Comment by nlwashington** [comment]
see #34 regarding genotype modeling.

**Comment by bryanlaraway** [comment]
@mbrush For modeling of the strain/background of a sequence alteration, is there an appropriate term to describe that relationship? Something akin to "in_strain/background" or "from_strain/background?" For example, for allele Itga3<tm1(KOMP)Vlcg>, the associated strain for that allele is C57BL/6NTac, while the the background of the actual mouse may be something different, such as 129.

**Comment by nlwashington** [comment]
@mbrush would "derives_from" be reasonable?

**Comment by mbrush** [comment]
@bryanlaraway @nlwashington  Well, the derives_from property from the Relation Ontology (RO_0001000) is meant to hold between material entities, so we wouldn't use this directly because alleles in our model are abstract things (generically dependent continuants in BFO-speak).  But I can create a new property in GENO called something like derives_from_background that we can use here. I will pass along an IRI for this property once I implement it.

**Comment by bryanlaraway** [comment]
I'm adding the different entities that have required mapping in MGI in case we can generalize them to utilities for use across different resources:

-Taxon: this seems one that should be generalized, or perhaps handled once NCBITaxon is added as a resource. If generalized, will need a way to spot if mapping from a resource results in errors, such as a discrepancy in name (Mus musculus/domesticus instead of just Mus musculus).

-Marker/gene type: These are often specific to the resource, but we could have a rolling mapping dictionary that we could add to as we encounter different terms in different resources.

-Zygosity: This one seems easy to generalize.

-Sequence alteration type: Similar to marker/gene type, we will encounter different terms in different resources, but can build a dictionary to standardize handling of terms that repeatedly pop up.

-Evidence ID/Evidence Label: I have the evidence code (EXP, IBA, NAS, etc.) and need to map it to the correct ID and label from the evidence ontology. Not sure if this should be generalized, or if it would be handled downstream with the loading of the evidence ontology. For completeness of my code now, I went ahead and mapped to both the label and the ID.

**Comment by bryanlaraway** [comment]
Regarding publications for the genotype to phenotype mapping, do we just want the pubmed IDs, or do we also want the MGI IDs and the Jnumbers?

**Comment by nlwashington** [comment]
absolutely, we want the J ids as sameAs ids.  you can follow the method I used for ZFINs pubs (although i think i might have used classes, but they should be individuals).

**Comment by bryanlaraway** [comment]
Another data modeling question: when processing the alleles from the gxd_allele_pair_view, I have the zygosity available along with allele1 and allele2. Am I correct in assuming that it is not necessary to include a triple indicating that allele1 is heterozygous with allele2, since we have the following three triples and the relationship can be inferred:
-vslc has_part allele1
-vslc has_part allele2
-vslc has_disposition mapped(allelestate)?

**Comment by bryanlaraway** [comment]
Currently there are a few data issues with a few tables/views ingested from MGI, consisting of additional tabs at the ends of some fields or spaces present in DOI IDs. Contacted MGI and received a response. DOIs should be corrected in next public release on 2/16/15. Tabs may take a little longer. Will have to implement a fix in the meantime, as discussed elsewhere.

**Comment by nlwashington** [comment]
i believe that the triples related to vslc above are the right ones. i don't think you need another relationship between the alleles.

**Comment by mbrush** [comment]
Re: the zygosity question, I agree that no additional triple is needed.  I am assuming the vslc has_disposition allelestate triple is  the one describing the zygosity of the vslc (using a zygosity term from geno as the object).  We have a specific object property to use here (GENO_0000608, has_zygosity), rather than has_disposition.

**Comment by bryanlaraway** [comment]
MGI is ready for code review. I'll create tickets for some of the known issues, including handling of the tables that have errors.

**Comment by nlwashington** [comment]
address many issues in code in commit b2b0936

**Comment by nlwashington** [comment]
@cmungall and @mellybelly most, but not all, mgi background strains have MGIids.  before i knew there were any MGIids, i was exposing the internal ids from the mgi tables for those strains.  i am hesitating to do that at all this time around.  i can get around it here by using an anonymous node, but it's awkward because it's not really anonymous!  we have a couple of options:
1.  expose the internal identifier
2.  create a monarchized identifier
3.  keep it anonymous and don't identify it.
what would you prefer?

**Comment by mellybelly** [comment]
I would send MGI a list and ask if they would create the IDs. If not tell them we will create them on their behalf OR use their internal table IDs (prefer we do the former).

**Comment by nlwashington** [comment]
as for the background genotype ids, for now i am creating them as blank nodes.

**Comment by nlwashington** [comment]
closing.  any changes to MGI should be placed in new ticket.
